pkg/registry/apis/provisioning/resources/dualwriter.go

```diff
@@ -317,3 +337,126 @@ func (r *DualReadWriter) authorizeCreateFolder(ctx context.Context, _ string) er
 	return apierrors.NewForbidden(FolderResource.GroupResource(), "",
 		fmt.Errorf("must be admin or editor to access folders with provisioning"))
 }
+
+func (r *DualReadWriter) deleteFolder(ctx context.Context, opts DualWriteOptions) (*ParsedResource, error) {
+	// if the ref is not the active branch, just delete the files from the branch
+	// do not delete the items from grafana itself
+	if opts.Ref != "" && opts.Ref != r.getConfiguredBranch() {
+		err := r.repo.Delete(ctx, opts.Path, opts.Ref, opts.Message)
+		if err != nil {
+			return nil, fmt.Errorf("error deleting folder from repository: %w", err)
+		}
+
+		return folderDeleteResponse(opts.Path, opts.Ref, r.repo.Config()), nil
+	}
+
+	// before deleting from the repo, first get all children resources to delete from grafana afterwards
+	treeEntries, err := r.repo.ReadTree(ctx, "")
```

Comment:
Maybe we extend this function or add another one in the repo interface to read subtrees (e.g. `ReadSubtree(context.Context, ref, path string) (Tree, error)`

Reply:
I'd personally vote to hold off until we know for sure that we need this outside of folder deletion -- long term, we'll want this logic to live in the grafana folder /apis.